pkgs/http/lib/src/multipart_file_io.dart

```diff
@@ -5,14 +5,14 @@
 import 'dart:io';
 
 import 'package:http_parser/http_parser.dart';
-import 'package:path/path.dart' as p;
 
 import 'byte_stream.dart';
 import 'multipart_file.dart';
 
 Future<MultipartFile> multipartFileFromPath(String field, String filePath,
     {String? filename, MediaType? contentType}) async {
-  filename ??= p.basename(filePath);
+  late var segments = File(filePath).uri.pathSegments;
```

Comment:
Can you go directly to the uri instead of through File?

Reply:
also curious why this is late?

Reply:
> Can you go directly to the uri instead of through File?
Yes! That is better.

> why this is late?
It only has to be computed if filename is `null`.

Reply:
oh I see just to make it lazy, sgtm then